include/databaseFunctions_blog_pic.php

```diff
@@ -1,58 +1,250 @@
+			$errors+=ValidateTextField('username', $errors);
+			$errors+=ValidateTextField('password', $errors);
+			if(sizeof($errors) == 0){
+				if(UserExists($_REQUEST['username'])){
+					if(GetUser($_REQUEST['username'])['password'] != $_REQUEST['password']){
+						$errors['match'] = 'incorrect password!';
 					}
+				}else{
+					$errors['DNE'] = 'user does not exist!';
 				}
-			</script>
+			}
+			if(sizeof($errors) == 0){
+				$_SESSION['username'] = $_REQUEST['username'];
+				header("Location: index.php");
+			}else{
+				if(isset($errors['username'])){
+					echo"<div class='required'>Please enter your username.</div>";
+				}
+				if(isset($errors['password'])){
+					echo"<div class='required'>Please enter your password.</div>";
+				}
+				if(isset($errors['match'])){
+					echo"<div class='required'>Your username and password do not match.</div>";
+				}
+				if(isset($errors['DNE'])){
+					echo"<div class='required'>There is no account associated with that username.</div>";
+				}
+			}
+		}
+		echo"<form method='post'>";
+		ShowTextField(true, 'username');
+		ShowPasswordField('password', 'password');
+		echo"<input type='submit' name = 'login' value='Login'>
+			</form>";
+	}
+
+	function IsLoggedIn(){
+		if(isset($_SESSION['username'])){
+			return true;
+		}
+		return false;
+	}
+
+	function PersonalHeading(){
+		echo"
+			<p class='personal'>hi ".$_SESSION['username']."</p>
 		";
+		Logout();
+	}
 
-		// if(isset($_REQUEST['delete'])){
-		// 	var_dump(GetPost($postID)['delKey']);
-		// 	echo"
-		// 		<script>
-		// 			alert('fdhldh');
-		// 			var key = prompt('Please enter delete key.');
-		// 			if(key == ".GetPost($postID)['delKey']."){
-		// 				".DeletePost($postID)."
-		// 			}
-		// 			else{
-		// 				alert('Wrong Delete Key! You do not have permission to delete this post.<br>');
-		// 			}
-		// 		</script>
-		// 	";
-		// }
-	}
-
-
-	//POST FUNCTIONS
-	function InsertBlogPost($author, $title, $body, $delKey){
+	function Logout(){
+		if(isset($_REQUEST['logout'])){
+			// echo"<script>
+			// 		alert('Are you sure you want to log out?')
+			// 	</script>
+			// ";
+			session_destroy();
+			header("Location: index.php");
+		}
+		echo"
+			<form>
+				<input class='personal' type='submit' name='logout' value='Log Out'>
+			</form>
+		";
+	}
+
+	function ValidDelete($postID){
+		if(isset($_SESSION['username'])){
+			if($_SESSION['username'] == GetPost($postID)['username']){
+				return true;
+			}
+			if(GetUser($_SESSION['username'])['userType'] == 'admin'){
+				return true;
+			}
+		}
+		return false;
+	}
+
+	//USER DATABASE FUNCTIONS
+	function AddNewUser($username, $password, $email){
 		$result = dbQuery("
-			INSERT INTO posts (author, title, body, isPic, delKey)
-			VALUES('$author', '$title', '$body', '0', '$delKey')
+			INSERT INTO users (username, password, userType, email)
+			VALUES('$username', '$password', 'regUser', '$email')
+		")->fetch();
+	}
+
+	function UserExists($username){
+		$result = dbQuery("
```

Comment:
I think this could be written as 

```
SELECT *
FROM users
WHERE username = '$username'
```

and then you just check to see if a row was returned. That's cleaner than using this sub-query I think.

Also, normally you want usernames to be case insensitive, so I'd suggest forcing them to be all uppercase or all lowercase for this comparison.